lib/Network/OAuth2/Server/API.hs

```diff
@@ -511,39 +503,33 @@ processAuthorizeGet ref user_id permissions response_type client_id' redirect sc
 
     -- Required: a supported ResponseType value.
     case response_type of
-        Nothing -> throwError $ ( Just redirect_uri
-                                , OAuth2Error InvalidRequest
-                                              (preview errorDescription "Response type is missing")
-                                              Nothing
-                                )
         Just ResponseTypeCode -> return ()
-        Just _ -> throwError ( Just redirect_uri
-                             , OAuth2Error InvalidRequest
-                                           (preview errorDescription "Invalid response type")
-                                           Nothing
-                             )
+        Just _  -> throwInvalidRequest redirect_uri "Invalid response type"
+        Nothing -> throwInvalidRequest redirect_uri "Response type is missing"
 
     -- Optional (but we currently require): requested scope.
     requested_scope <- case scope' of
-        Nothing ->
-            throwError $ ( Just redirect_uri
-                         , OAuth2Error InvalidRequest
-                                         (preview errorDescription "Scope is missing")
-                                         Nothing
-                         )
-        Just requested_scope ->
-            if requested_scope `compatibleScope` permissions
-                then return requested_scope
-                else throwError ( Just redirect_uri
-                                , OAuth2Error InvalidScope
-                                              (preview errorDescription "Invalid scope")
-                                              Nothing
-                                )
+        Nothing -> throwInvalidRequest redirect_uri "Scope is missing"
+        Just requested_scope
+            | requested_scope `compatibleScope` permissions -> return requested_scope
+            | otherwise -> throwError
```

Comment:
Had that originally, but was only used in one place, so I removed it. I'll add it back.